linux/net/rina/ipcps/shim-eth-vlan.c

```diff
@@ -300,13 +301,8 @@ static string_t * create_vlan_interface_name(string_t * interface_name,
         string_t * complete_interface;
         size_t     length;
 
-        if (!interface_name)
-                return NULL;
-
-        if (!vlan_id_is_ok(vlan_id)) {
-                LOG_ERR("Wrong vlan-id %d", vlan_id);
-                return NULL;
-        }
+        ASSERT(interface_name);
+        ASSERT(vlan_id_is_ok(vlan_id));
```

Comment:
This will break the regression tests. It makes sense not to assert in this function, but to do checks on the value of the VLAN.